src/lib/support/BufferWriter.cpp

```diff
@@ -30,7 +30,7 @@ BufferWriter & BufferWriter::Put(const void * buf, size_t len)
 {
     size_t available = Available();
 
-    if (available > 0)
+    if (available > 0 && buf != nullptr)
```

Comment:
Seems ok, but would checking for len > 0 be better?

Reply:
Good point, I have updated it. Thanks!